### PR TITLE
Correctly detect relative Uri in package parts

### PR DESCRIPTION
**Pull request by igitur**
I'm using Documentformat.OpenXML to load a spreadsheet document. The package parts have relative Uris, e.g. /xl/style.xml. Currently Mono will parse that Uri as an absolute Uri (hence to the filesystem). A workaround is needed to treat it as a relative Uri.

Refer to http://www.mono-project.com/docs/faq/known-issues/urikind-relativeorabsolute/ for description of recommended workaround.

### CODE REVIEW

**Comment by igitur** [comment]
Or should this check be done only for systems on which the path separator is a forward slash?

File: mcs/class/WindowsBase/Consts.cs [hunk 1]
```diff
@@ -0,0 +1,124 @@
+//
```
Comment (marek-safar) [inline mcs/class/WindowsBase/Consts.cs]: This file should not be copied

File: mcs/class/WindowsBase/System.IO.Packaging/PackagePart.cs [hunk 2]
```diff
@@ -165,10 +165,13 @@ void LoadRelationships (Dictionary<string, PackageRelationship> relationships, S
...
 				if (node.Attributes["TargetMode"] != null)
 					mode = (TargetMode) Enum.Parse (typeof(TargetMode), node.Attributes ["TargetMode"].Value);
-				
-				CreateRelationship (new Uri (node.Attributes["Target"].Value.ToString(), UriKind.RelativeOrAbsolute),
-				                    mode,
-				                    node.Attributes["Type"].Value.ToString (),
+
+				// Workaround for Mono relative paths
+				// http://www.mono-project.com/docs/faq/known-issues/urikind-relativeorabsolute/
+				var kind = node.Attributes["Target"].Value.ToString()?.StartsWith ("/") == true ? UriKind.Relative : UriKind.RelativeOrAbsolute;
```
Comment (marek-safar) [inline mcs/class/WindowsBase/System.IO.Packaging/PackagePart.cs]: Use `var kind = (UriKind) 300;` instead

Comment (marek-safar) [inline mcs/class/WindowsBase/System.IO.Packaging/PackagePart.cs]: Could you also add a test for this

**Comment by igitur** [comment]
OK, will do so.

**Comment by mattzink** [comment]
Why not just set "useDotNetRelativeOrAbsolute" field as described in the link you provided? If you want leading forward slash to mean relative, you would want that behavior globally, not just in PackagePart, right?

**Comment by igitur** [comment]
Sorry for the delay. I made the changes as requested, including adding a unit test.

Do you think the same logic should be applied to [Package.cs#L325](https://github.com/mono/mono/blob/master/mcs/class/WindowsBase/System.IO.Packaging/Package.cs#L325) too?

**Comment by igitur** [comment]
@mattzink As the documentation states, `useDotNetRelativeOrAbsolute` should be used with care as it will treat all Uri's with a forward slash as relative. This might break other 3rd party libraries.

**Comment by marek-safar** [comment]
build

**Pull request merged**